packages/starknet-snap/src/utils/formatter-utils.ts

```diff
@@ -16,6 +16,28 @@ export const hexToString = (hexStr) => {
   return str;
 };
 
+/**
+ * Normalizes a StarkNet hex string (address or transaction hash) to 66 characters,
+ * including the '0x' prefix. Ensures the hex value is padded with leading zeros
+ * to meet StarkNet's 64-character requirement (excluding the prefix).
+ *
+ * @param hexString - the input hex string to normalize.
+ * @returns a normalized 66-character hex string with the '0x' prefix and added leading zeros if necessary.
+ */
+export const normalizeStarknetHex = (hexString: string) => {
+  if (!hexString.startsWith('0x')) {
+    throw new Error('Invalid input: Hex value must start with "0x"');
+  }
+
+  const hexWithoutPrefix = hexString.slice(2); // Remove '0x'
+
+  if (hexWithoutPrefix.length > 64) {
+    throw new Error('Invalid input: Hex value is longer than 64 characters');
+  }
+
+  return `0x${hexWithoutPrefix.padStart(64, '0')}`; // Pad to 64 chars + '0x'
+};
```

Comment:
i guess what you try to do is verify if a string is hex string, and its length cant exceed 64

may be just use validateAndParseAddress from starknet.js, it trim, add hex, case in sensitive

```
 * const address = "0x90591d9fa3efc87067d95a643f8455e0b8190eb8cb7bfd39e4fb7571fdf";
 * const result = validateAndParseAddress(address);
 * // result = "0x0000090591d9fa3efc87067d95a643f8455e0b8190eb8cb7bfd39e4fb7571fdf"
 ```

Reply:
Did not know of validateAndParseAddress will also replace other manual padding we were doing with this